### PR TITLE
search detail ui and fetch data

### PR DESCRIPTION
**Pull request by imemoasia**
## Related Tickets

- [#28794](https://dev.framgia.com/redmine/issues/28794)
- [#28790](https://dev.framgia.com/redmine/issues/28790)

## What's this PR do ?
- create search detail ui
- fetch data search

### CODE REVIEW

File: Unsplash/Unsplash/Scene/DetailPhotoCollecton/DetailPhotoCollectionVC.swift [hunk 1]
```diff
@@ -51,7 +50,10 @@ class DetailPhotoCollectionVC: UIViewController {
             }
         }
     }
-    
+    override func viewWillAppear(_ animated: Bool) {
+        tabBarController?.tabBar.isHidden = true
```
Comment (datdt-1935) [inline Unsplash/Unsplash/Scene/DetailPhotoCollecton/DetailPhotoCollectionVC.swift:55]: thêm super

File: Unsplash/Unsplash/Scene/DetailSearch/DetailSearchViewController.swift [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    
+    override func viewWillAppear(_ animated: Bool) {
+        tabBarController?.tabBar.isHidden = true
+        let backButton = UIBarButtonItem()
+        backButton.title = "Search"
+        self.navigationController?.navigationBar.topItem?.backBarButtonItem = backButton
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        tabBarController?.tabBar.isHidden = false
```
Comment (datdt-1935) [inline Unsplash/Unsplash/Scene/DetailSearch/DetailSearchViewController.swift:90]: thêm super

File: Unsplash/Unsplash/Repositories/SearchRepositories.swift [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+//  SearchRepositories.swift
+//  Unsplash
+//
+//  Created by Nha Pham on 9/3/20.
+//  Copyright © 2020 Nha Pham. All rights reserved.
+//
+
+import Foundation
+
+final class SearchRepositories: BaseRepository {
```
Comment (datdt-1935) [inline Unsplash/Unsplash/Repositories/SearchRepositories.swift]:
```suggestion
final class SearchRepository: BaseRepository {
```

File: Unsplash/Unsplash/Services/Request/SearchRequest.swift [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            URLQueryItem(name: "per_page", value: "\(elementPerPage)"),
+            URLQueryItem(name: "order_by", value: "\(orderBy)")
+        ]
+        super.init(baseUrl: URls.API.baseUrl, path: path, urlComponents: parameters)
+    }
+    
+    init(search type: SearchType, query: String, page: Int, elementPerPage: Int) {
+           let path = "\(URls.API.search)/\(type.rawValue)"
+           let parameters = [
+               URLQueryItem(name: "client_id", value: APIKey.apiKey),
```
Comment (datdt-1935) [inline Unsplash/Unsplash/Services/Request/SearchRequest.swift]: cái này có trong base request rồi mà nhỉ?

Comment (imemoasia) [reply Unsplash/Unsplash/Services/Request/SearchRequest.swift]: dạ vâng em quên mất ạ

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**